src/test/spec/unified_runner/operation.rs

```diff
@@ -566,10 +568,19 @@ impl TestOperation for Aggregate {
         async move {
             let result = match &self.session {
                 Some(session_id) => {
-                    let entity = test_runner.entities.get(id).unwrap().clone();
+                    enum AggregateEntity {
```

Comment:
Are these changes necessary? It looks functionally pretty similar to the previous implementation.

Reply:
Yup, it's needed - pre-change, the entity retrieved was unconditionally cloned; with this change, only a collection or database is cloned, other entities are formatted for later debug output.